File: db/csharp/SalesManagement.Domain/Models/CreditBalance.cs
namespace SalesManagement.Domain.Models;

/// <summary>
/// 与信残高データ
/// </summary>
public class CreditBalance
{
    /// <summary>
    /// 取引先コード
    /// </summary>
    public string CompanyCode { get; set; } = string.Empty;

    /// <summary>
    /// 受注残高
    /// </summary>
    public decimal OrderBalance { get; set; }

    /// <summary>
    /// 債権残高
    /// </summary>
    public decimal ReceivableBalance { get; set; }

    /// <summary>
    /// 債務残高
    /// </summary>
    public decimal PayableBalance { get; set; }

    /// <summary>
    /// 作成日時
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 作成者名
    /// </summary>
    public string CreatedBy { get; set; } = string.Empty;

    /// <summary>
    /// 更新日時
    /// </summary>
    public DateTime UpdatedAt { get; set; }

    /// <summary>
    /// 更新者名
    /// </summary>
    public string UpdatedBy { get; set; } = string.Empty;
}

File: db/fsharp/SalesManagement.Domain/Models/CreditBalance.fs
namespace SalesManagement.Domain.Models

open System

/// 与信残高データ
[<CLIMutable>]
type CreditBalance = {
    CompanyCode: string          // 取引先コード（PK）
    OrderBalance: decimal        // 受注残高（受注したが未出荷の金額）
    ReceivableBalance: decimal   // 債権残高（出荷したが未入金の金額）
    PayableBalance: decimal      // 債務残高（仕入したが未支払の金額）
    CreatedAt: DateTime          // 作成日時
    CreatedBy: string option     // 作成者名
    UpdatedAt: DateTime          // 更新日時
    UpdatedBy: string option     // 更新者名
}
